eng/scripts/Attest-CPEX-KPIs.ps1

```diff
@@ -0,0 +1,314 @@
+$KPI_ID_Mgmt_Public_Preview = "84715402-4f3c-4dca-b330-f05206abaec5"
+$KPI_ID_Data_Public_Preview = "ad70777b-a1f5-4d77-8926-5c466d7a214d"
+$KPI_ID_Mgmt_GA = "210c095f-b3a2-4cf4-a899-eaab4c3ed958"
+$KPI_ID_Data_GA = "da768dff-8f90-4999-ad3a-adcd790911f3"
+
+$KPI_ID_TO_TITLE = @{
+    $KPI_ID_Onboarding             = "Onboarding"
+    $KPI_ID_Mgmt_Private_Preview   = "Management Plane - Private Preview API readiness"
+    $KPI_ID_Data_Private_Preview   = "Data Plane - Private Preview API readiness"
+    $KPI_ID_Mgmt_Public_Preview    = "Management Plane - API readiness and Beta SDK release"
+    $KPI_ID_Data_Public_Preview    = "Data Plane - API readiness and Beta SDK release"
+    $KPI_ID_Mgmt_GA                = "Management Plane - API readiness and Stable SDK release"
+    $KPI_ID_Data_GA                = "Data Plane - API readiness and Stable SDK release"
+}
+
+$SERVICE_TREE_URL = "https://microsoftservicetree.com/products/"
+
+$successfulAttestations = @{}
+$failedAttestations = @()
+
+function InvokeKustoCommand($command) {
+    try {
+        $clusterUri = "https://azsdk-cpex-attestation.westus2.kusto.windows.net"
+        $databaseName = "CPEX_Attestation_DB"
+        $accessToken = az account get-access-token --resource "https://api.kusto.windows.net" --query "accessToken" --output tsv
+        $headers = @{ Authorization="Bearer $accessToken" }
+        $body = @{ csl = $command; db = $databaseName } | ConvertTo-Json -Depth 3
+        Invoke-RestMethod -Uri "$clusterUri/v1/rest/mgmt" -Headers $headers -Method Post -Body $body -ContentType "application/json"
+    } catch {
+        Write-Error "Failed to invoke Kusto command: $command"
+        Write-Error "Exception message: $($_.Exception.Message)"
+        throw "Terminating due to failure in invoking kusto command"
+    }
+
+}
+
+function AddAttestationEntry($targetId, $actionItemId, $status, $targetType, $url, $productName) {
+    $command = @"
+.append $TableName <|
+print
+    ActionItemId = "$actionItemId",
+    TargetId = "$targetId",
+    TargetType = "$targetType",
+    Status = int($status),
+    CreatedTime = datetime(now),
+    EvidenceUrl = "$url"
+"@
+
+    Write-Host "Adding attestation entry for product [$targetId], with status [$status] for KPI action id [$actionItemId]."
+    InvokeKustoCommand $command
+    Write-Host "Added attestation entry for product [$targetId], with status [$status] for KPI action id [$actionItemId]."
+
+    if (-not $successfulAttestations.ContainsKey($actionItemId)) {
+        $successfulAttestations[$actionItemId] = @()
+    }
+
+    $successfulAttestations[$actionItemId] += @{
+        productId = $targetId
+        productName = $productName
+        status = if ($status -eq $COMPLETED) { "Completed" } else { "Not Applicable"}
+    }
+}
+
+function SendEmailNotification($emailTo, $emailCC, $emailSubject, $emailBody) {
+    try {
+        $emailerUri = $env:azure_sdk_emailer_uri
```

Comment:
This may not work from pipeline. You will need to pass the URI as a param to this script from pipeline yaml. URI is already present in the keyvault. I have added a variable group in Azure DevOps to access this KV so you need to pass `$(AzureSDKEmailerSasURL)` as value from pipeline.